### PR TITLE
Add new version syntax based on yml files to the pipeline

### PR DESCRIPTION
**Pull request by JoseEspinosa**
This PR implements the new way of outputing software versions. Software version(s) are now reported for every module imported during a given pipeline execution that is emitted in a `versions.yml` file.
Other minor changes:

*  `params.skip_mirdeep` switch is implemented on `smrnaseq.nf` instead of inside the modules involved in `MIRDEEP2 ` subworkflow.
* Remove hostnames to avoid error in json schema validation in new Nextflow releases (see here nf-core/tools#1304)
* Update mirbase urls
* Update singularity container of `modules/local/bowtie_map_mirna.nf` to match docker one.

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] Make sure your code lints (`nf-core lint .`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [x] `CHANGELOG.md` is updated.

### CODE REVIEW

File: modules/local/bowtie_map_mirna.nf [hunk 1]
```diff
@@ -1,15 +1,13 @@
...
-options        = initOptions(params.options)
 
 process BOWTIE_MAP_SEQ {
-    label 'process_medium'
     tag "$meta.id"
-
+    label 'process_medium'
     publishDir "${params.outdir}",
-        mode: params.publish_dir_mode,
+        mode: params.publish_dir_mode, //TODO check for suffix, is working? needs initOptions?
```
Comment (FriederikeHanssen) [inline modules/local/bowtie_map_mirna.nf:10]: Is this TODO still relevant? If so do you plan on adressing it in a different PR?

File: modules/local/bowtie_map_mirna.nf [hunk 2]
```diff
@@ -24,16 +22,14 @@ process BOWTIE_MAP_SEQ {
...
-    tuple val(meta), path('unmapped/*fq.gz') , emit: unmapped
-    path "*.version.txt" , emit: versions
+    tuple val(meta), path("*bam")           , emit: bam
+    tuple val(meta), path('unmapped/*fq.gz'), emit: unmapped
+    path "versions.yml"                     , emit: versions
 
     script:
-    def software = getSoftwareName(task.process)
-    def process_name = task.process.tokenize(':')[-1]
+    // def process_name = task.process.tokenize(':')[-1] //TODO check if is still needed
```
Comment (FriederikeHanssen) [inline modules/local/bowtie_map_mirna.nf:30]: same here

File: modules/local/seqcluster_collapse.nf [hunk 3]
```diff
@@ -1,5 +1,5 @@
 // Import generic module functions
-include { saveFiles; initOptions; getSoftwareName } from './functions'
+include { saveFiles; initOptions; getSoftwareName; getProcessName } from './functions'
```
Comment (FriederikeHanssen) [inline modules/local/seqcluster_collapse.nf:2]: is initOptions still needed here?

File: modules/local/trimgalore.nf [hunk 4]
```diff
@@ -1,5 +1,5 @@
 // Import generic module functions
-include { saveFiles; initOptions; getSoftwareName } from './functions'
+include { saveFiles; initOptions; getSoftwareName; getProcessName } from './functions'
```
Comment (FriederikeHanssen) [inline modules/local/trimgalore.nf:2]: same here

**Review by FriederikeHanssen — COMMENTED** [review]
Nice looks good. Just a couple of things about open TODO statements and having `initOptions` left here and there.

File: modules/local/trimgalore.nf (hunk 4 above)
Comment (JoseEspinosa) [reply modules/local/trimgalore.nf]: Addressed here ab5e61a

File: modules/local/bowtie_map_mirna.nf (hunk 1 above)
Comment (JoseEspinosa) [reply modules/local/bowtie_map_mirna.nf]: I updated the publishDir of all the bowtie local modules here: a5ae9fc

File: modules/local/bowtie_map_mirna.nf (hunk 2 above)
Comment (JoseEspinosa) [reply modules/local/bowtie_map_mirna.nf]: I updated the publishDir of all the bowtie local modules here: a5ae9fc

**Review by FriederikeHanssen — APPROVED** [review]
lgtm

**Comment by JoseEspinosa** [comment]
I think I have addressed now all @FriederikeHanssen suggestions (very nicely spotted :smile: ) + I updated the files from mirbase in the `nextflow.config` as discussed on slack.

**Review by lpantano — APPROVED** [review]

**Pull request merged**